logger/syslogish/server.go

```diff
@@ -0,0 +1,170 @@
+package syslogish
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/deis/deis/logger/drain"
+	"github.com/deis/deis/logger/storage"
+)
+
+var appRegex *regexp.Regexp
+
+func init() {
+	appRegex = regexp.MustCompile(`^.* ([-_a-z0-9]+)\[[a-z0-9-_\.]+\].*`)
+}
+
+// Server implements a UDP-based "syslog-like" server.  Like syslog, as described by RFC 3164, it
+// expects that each packet contains a single log message and that, conversely, log messages are
+// encapsulated in their entirety by a single packet, however, no attempt is made to parse the
+// messages received or validate that they conform to the specification.
+type Server struct {
+	conn           net.PacketConn
+	listening      bool
+	queue          chan string
+	storageAdapter storage.Adapter
+	drain          drain.LogDrain
+	adapterMutex   sync.RWMutex
+	drainMutex     sync.RWMutex
+}
+
+// NewServer returns a pointer to a new Server instance.
+func NewServer(bindHost string, bindPort int) (*Server, error) {
+	addr, err := net.ResolveUDPAddr("udp", fmt.Sprintf("%s:%d", bindHost, bindPort))
+	if err != nil {
+		return nil, err
+	}
+	c, err := net.ListenUDP("udp", addr)
+	if err != nil {
+		return nil, err
+	}
+	return &Server{conn: c, queue: make(chan string, 5)}, nil
```

Comment:
So this is an interesting thing... while I've refactored most of the code for easier maintenance, some of the mechanics in terms of how this component actually works are inspired by / true to the original implementation.  The queue depth of 5 came straight out of there.

That being said, I do not object to increasing it.